### PR TITLE
RSDK-3765: Flutter SDK Gantry API changes

### PR DESCRIPTION
**Pull request by martha-johnston**

### CODE REVIEW

File: lib/src/components/gantry/gantry.dart [hunk 1]
```diff
@@ -9,8 +9,11 @@ abstract class Gantry extends Resource {
   /// Get the position of the axes in millimeters
   Future<List<double>> position({Map<String, dynamic>? extra});
 
-  /// Move the gantry to a new position
-  Future<void> moveToPosition(List<double> positions, {Map<String, dynamic>? extra});
+  /// Move the gantry to a new position at the requested speeds
```
Comment (njooma) [inline lib/src/components/gantry/gantry.dart:12]: Could you add the units here? Seems to be the only comment missing them

File: test/unit_test/components/gantry_test.dart [hunk 2]
```diff
@@ -70,19 +77,27 @@ void main() {
       expect(await gantry.position(), [0, 0, 0]);
       expect(gantry.isStopped, true);
 
-      await gantry.moveToPosition([1, 2, 3]);
+      await gantry.moveToPosition([2, 4, 6], [4, 8, 12]);
+      expect(gantry.positions, [2, 4, 6]);
```
Comment (njooma) [inline test/unit_test/components/gantry_test.dart:81]: Add another test here for checking speed

**Review by njooma — CHANGES_REQUESTED** [review]
Looking good! Just a few comments

File: test/unit_test/components/gantry_test.dart [hunk 3]
```diff
@@ -150,7 +165,8 @@ void main() {
         final client = GantryServiceClient(channel);
         final request = MoveToPositionRequest()
           ..name = name
-          ..positionsMm.addAll([1, 2, 3]);
+          ..positionsMm.addAll([1, 2, 3])
+          ..speedsMmPerSec.addAll([4, 8, 12]);
         await client.moveToPosition(request);
 
         expect(gantry.positions, [1, 2, 3]);
```
Comment (clintpurser) [inline test/unit_test/components/gantry_test.dart:172]: similar to above, add another check for speeds here.

File: test/unit_test/components/gantry_test.dart [hunk 4]
```diff
@@ -226,7 +244,7 @@ void main() {
         expect(gantry.isStopped, true);
 
         final client = GantryClient(name, channel);
-        await client.moveToPosition([1, 2, 3]);
+        await client.moveToPosition([1, 2, 3], [4, 8, 12]);
 
         expect(gantry.positions, [1, 2, 3]);
```
Comment (clintpurser) [inline test/unit_test/components/gantry_test.dart:249]: also add check for speeds here

**Review by clintpurser — CHANGES_REQUESTED** [review]
Overall looks good, pointed out a few more spots i'd add checks for speed.

**Comment by clintpurser** [comment]
Great job jumping into some dart @martha-johnston !!!

**Review by benjirewis — APPROVED** [review]
Nice! Great job on the dart 🎯 @martha-johnston (not that I know any dart lol); thank you!

**Review by clintpurser — APPROVED** [review]
LGTM

**Review by njooma — APPROVED** [review]

**Pull request merged**